crates/circuit/src/dag_circuit.rs

```diff
@@ -4996,6 +4993,44 @@ impl DAGCircuit {
         Ok(new_node)
     }
 
+    fn get_classical_resources(
+        &mut self,
+        py: Python,
+        instr: PackedInstruction,
+    ) -> PyResult<DAGNodeInfo> {
+        let (all_clbits, vars): (Vec<Clbit>, Option<Vec<Var>>) = {
+            if self.may_have_additional_wires(py, &instr) {
+                let mut clbits: HashSet<Clbit> =
+                    HashSet::from_iter(self.cargs_interner.get(instr.clbits).iter().copied());
+                let (additional_clbits, additional_vars) =
+                    self.additional_wires(py, instr.op.view())?;
+                for clbit in additional_clbits {
+                    clbits.insert(clbit);
+                }
+                let additional_vars = additional_vars
+                    .into_iter()
+                    .map(|var| self.vars.find(&var.bind(py).into()).unwrap())
+                    .collect();
+                (clbits.into_iter().collect(), Some(additional_vars))
+            } else {
+                (self.cargs_interner.get(instr.clbits).to_vec(), None)
+            }
+        };
+
+        // Get the correct qubit indices
+        let qubits_id = instr.qubits;
+
+        let result = DAGNodeInfo {
+            all_clbits,
+            vars,
+            node: self.dag.add_node(NodeType::Operation(instr)),
+            qubits_id,
+        };
+
+        // Insert op-node to graph.
+        Ok(result)
```

Comment:
I was thinking that this stuff should not be part of `get_classical_resources`. This method should IMO just return `(all_clbits, vars)` and should probably take `instr` by reference.

(I don't think we need `DAGNodeInfo` at all anymore if we do that.)